src/codegen.rs

```diff
@@ -125,6 +134,86 @@ impl<'ink> CodeGen<'ink> {
             variable_generator.generate_global_variables(dependencies, &self.module_location)?;
         index.merge(llvm_gv_index);
 
+        // Build our GOT layout here. We need to find all the names for globals, programs, and
+        // functions and assign them indices in the GOT, taking into account prior indices.
+        let program_globals =
+            global_index.get_program_instances().into_iter().fold(Vec::new(), |mut acc, p| {
+                acc.push(p.get_name().to_owned());
+                acc.push(p.get_qualified_name().to_owned());
+                acc.push(format!("{}_instance", p.get_name()));
```

Comment:
This is way too many names, and a workaround because we did not yet have the time to research how to get certain globals, such as `init` variables for program instances. This causes a lot of unnecessary entries in the custom GOT, hence certain hacks in the runtime to only fetch a variable's address if it is present in the binary instead of trusting the custom GOT blindly.